### PR TITLE
sycl-bench added & ascii bar chart

### PR DESCRIPTION
**Pull request by mateuszpn**

### CODE REVIEW

File: .github/workflows/bench_publish.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Benchmarks Publish
```
Comment (pbalcer) [inline .github/workflows/bench_publish.yml:1]: please remove this.

File: scripts/benchmarks/benches/compute.py [hunk 2]
```diff
@@ -18,6 +18,8 @@ def __init__(self, directory):
         return
 
     def setup(self):
+        # self.bins = os.path.join(self.directory, 'compute-benchmarks-build', 'bin')
```
Comment (pbalcer) [inline scripts/benchmarks/benches/compute.py:21]: please remove the commented out parts

File: scripts/benchmarks/benches/syclbench.py [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+                if not row[0].startswith('#'):
+                    res_list.append(
+                        Result(label=row[0], 
+                               value=float(row[12]) * 1000, # convert to ms 
+                               command=command, 
+                               env=env_vars, 
+                               stdout=row))
+                            #    stdout=result))
+            
+            # median_list = []
```
Comment (pbalcer) [inline scripts/benchmarks/benches/syclbench.py:98]: ditto

File: scripts/benchmarks/output.py [hunk 4]
```diff
@@ -112,30 +85,140 @@ def generate_summary_table(chart_data: dict[str, list[Result]]):
...
+        s = l.label
+        prefix = re.match(r'^[^_\s]+', s)[0]
+        grouped_objects[prefix].append(l)
+
+    grouped_objects = dict(grouped_objects)
+
+    if mean_cnt > 0:
+        global_mean = global_product ** (1/mean_cnt)      
+        summary_line = f"Total {mean_cnt} benchmarks in mean. "
+        summary_line += "\n" + f"Geomean {global_mean*100:.3f}%. \nImproved {improved} Regressed {regressed} (treshold {epsilon*100:.2f}%)"
```
Comment (pbalcer) [inline scripts/benchmarks/output.py:170]:
```suggestion
        summary_line += "\n" + f"Geomean {global_mean*100:.3f}%. \nImproved {improved} Regressed {regressed} (threshold {epsilon*100:.2f}%)"
```

File: scripts/benchmarks/output.py [hunk 5]
```diff
@@ -112,30 +85,140 @@ def generate_summary_table(chart_data: dict[str, list[Result]]):
...
+
+    diff_values = [l.diff for l in sorted_detailed_list if l.diff is not None]
+
+    if len(diff_values) > 0:
+        max_diff = max(max(diff_values) - 1, 1 - min(diff_values))
+
+        for l in sorted_detailed_list:
+            if l.diff != None:
+                l.row += f" | {(l.diff - 1)*100:.2f}%"
+                epsilon = 0.005
```
Comment (pbalcer) [inline scripts/benchmarks/output.py:128]: Make this a configurable variable, and set it at ~5% by default.
We clearly have some work to do to stabilize the benchmarks, but let's do that later.

File: scripts/benchmarks/output.py [hunk 6]
```diff
@@ -112,30 +85,140 @@ def generate_summary_table(chart_data: dict[str, list[Result]]):
...
+
+
+    sorted_detailed_list = sorted(output_detailed_list, key=lambda x: (x.diff is not None, x.diff), reverse=True)
+
+    diff_values = [l.diff for l in sorted_detailed_list if l.diff is not None]
+
+    if len(diff_values) > 0:
+        max_diff = max(max(diff_values) - 1, 1 - min(diff_values))
+
+        for l in sorted_detailed_list:
```
Comment (pbalcer) [inline scripts/benchmarks/output.py:125]: don't use `l` or `1` for variable names.

**Review by pbalcer — COMMENTED** [review]
lgtm, just a few nits.

File: scripts/benchmarks/output.py (hunk 5 above)
Comment (mateuszpn) [reply scripts/benchmarks/output.py]: --epsilon added as an option, with 0.005 default

**Pull request merged**